### PR TITLE
[RLlib] from_config util method for framework agnostic components; start moving RLlib tests into Bazel.

### PR DESCRIPTION
**Pull request by sven1977**
In preparation for Exploration components (which will be framework-agnostic).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: rllib/utils/tests/test_framework_independent_components.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+from abc import ABCMeta, abstractmethod
```
Comment (ericl) [inline rllib/utils/tests/test_framework_independent_components.py]: Can you add this test to bazel? (actually are there other tests we aren't running anywhere?)

**Review by ericl — CHANGES_REQUESTED** [review]

File: rllib/utils/tests/test_framework_independent_components.py (hunk 1 above)
Comment (sven1977) [reply rllib/utils/tests/test_framework_independent_components.py]: Sure ...

Comment (sven1977) [reply rllib/utils/tests/test_framework_independent_components.py]: Yes: agents/ppo/test/test.py was never run.
I moved it to the new BUILD.bazel file in /rllib/

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
image

Nice!

**Comment by ericl** [comment]
OSX test looks like it failed due to a network failure.

**Pull request merged**